pkg/faucet/faucet.go

```diff
@@ -0,0 +1,286 @@
+package faucet
```

Comment:
This file needs a bit of a high level summary. 

I don't really know what to expect or what it does.

Reply:
E.g. Is this a `faucet_rest_api_server`?